packages/lib/src/vis/utils.ts

```diff
@@ -66,6 +68,11 @@ const adaptedLogTicksThreshold = scaleLinear({
   range: [0.8, 1.4],
 });
 
+export function createScale<
+  V extends VisScaleType,
+  S extends ExtractScaleType<V>
+>(scaleType: V, config: ScaleConfig<S>): Scale<S>;
```

Comment:
Couldn't do it without a function override. It's simpler to keep the main function signature "generic" to avoid resorting to workarounds inside the function body, and to enforce a stricter signature for the caller with an override.